pkg/docker/config/config.go

```diff
@@ -143,8 +153,24 @@ func GetAllCredentials(sys *types.SystemContext) (map[string]types.DockerAuthCon
 // The homeDir parameter should always be homedir.Get(), and is only intended to be overridden
 // by tests.
 func getAuthFilePaths(sys *types.SystemContext, homeDir string) []authPath {
+	runningInSystemd := os.Getenv("INVOCATION_ID") != ""
```

Comment:
Can you elaborate on why systemd is being excluded? Running Podman in systemd by means of Quadlet is widely used, so I want to make sure to understand the reasoning behind.

Reply:
See below, it isn't about excluding systemd cases, the current logic just changes the priority order for the search to search the system paths *first* - but in order to avoid breaking people today that are e.g. writing to `/root/.config` we still need to search those paths too.

The ordering logic here is IMO up for debate of course.

Reply:
Can you elaborate why the order should be different when running in systemd?  Certainly, we need to continue using the existing paths but I'd expect .d files to behave consistently.

Reply:
After this we'll have two big sources of auth: "user config" and "system config".

My thought was that when running in systemd we prefer system config over root's (or whatever user identity) first. This would avoid e.g. SELinux denials from system services trying to access root's home directory etc. It would be a fully backwards compatible change because the system wide paths wouldn't exist.